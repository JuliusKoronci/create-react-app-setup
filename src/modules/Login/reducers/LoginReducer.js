// @flow
import loginActions  from '../constants/LoginConstants';
import { APIResponse, ReduxApiAction } from '../../../FlowTypes';

const handleRequest = (state) => {return { ...state, request: true }};
const handleSuccess = (state, action) => {
	return {
		...state,
		request: false,
		loaded: true,
		success: true,
		data: action.response
	}
};
const handleError = (state, action) => {
	return {
		...state,
		request: false,
		loaded: true,
		success: false,
		error: action.response,
	}
};

const actions = {
	[loginActions.LOGIN_REQUEST]: handleRequest,
	[loginActions.LOGIN_ERROR]: handleError,
	[loginActions.LOGIN_SUCCESS]: handleSuccess,
};

export const initialState: APIResponse = {
	request: false,
	success: false,
	error: false,
	loaded: false,
	data: {
		token: '',
		username: '',
	}
};

const loginReducer = (state?: APIResponse = initialState, action: ReduxApiAction) => {
	try {
		actions[action.type](state);
	} catch (e) {
		return state;
	}
};

export default loginReducer;

