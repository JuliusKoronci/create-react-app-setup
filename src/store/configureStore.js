import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import createLogger from 'redux-logger';
import createHistory from 'history/createBrowserHistory';
import rootReducer from './rootReducer';
import DevTools from '../modules/DevTools';

export const history = createHistory();
const routerMiddlewareWithHistory = routerMiddleware(history);

function applyMiddlewareWithDevTool(...middleware) {
	if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
		return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(
			applyMiddleware(...middleware)
		);
	}

	return compose(
		applyMiddleware(...middleware),
		DevTools.instrument()
	);

}

const createDevStore = createStore(
	rootReducer,
	applyMiddlewareWithDevTool(routerMiddlewareWithHistory, thunk, createLogger)
);

const createProdStore = createStore(
	rootReducer,
	applyMiddleware(routerMiddlewareWithHistory, thunk)
);
const env = process.env.NODE_ENV;
const store = env === 'development' ? createDevStore : createProdStore;

if (module.hot) {
	module.hot.accept('./rootReducer', () => {
		const nextRootReducer = require('./rootReducer').default;
		store.replaceReducer(nextRootReducer);
	});
}

export default store;
