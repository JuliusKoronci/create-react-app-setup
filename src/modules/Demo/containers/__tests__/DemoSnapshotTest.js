import React from 'react';
import Demo from '../Demo';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { initialState } from '../../reducers/DemoReducer';

const middlewares = [];
const mockStore = configureStore(middlewares);

// Initialize mockstore with empty state
const store = mockStore({ demo: initialState });

const wrapper = shallow(
	<Demo store={store} />
);
describe('(Component) Demo', () => {
	it('Should match snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

