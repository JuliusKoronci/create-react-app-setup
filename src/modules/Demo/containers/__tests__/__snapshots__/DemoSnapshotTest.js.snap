exports[`(Component) Demo Should match snapshot 1`] = `
<Demo
  createEntry={[Function]}
  deleteEntry={[Function]}
  demo={
    Object {
      "list": Array [
        Object {
          "desc": "This is a demo test whatever json object",
          "id": 1,
          "label": "Test",
        },
      ],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;
